pkg/provider/vault/auth.go

```diff
@@ -160,6 +161,16 @@ func checkToken(ctx context.Context, token util.Token) (bool, error) {
 	if tokenType == "batch" {
 		return false, nil
 	}
+	ttl, ok := resp.Data["ttl"]
+	if !ok {
+		return false, fmt.Errorf("no TTL found in response")
+	}
```

Comment:
from vault docs, it isn't clear for me that this field is mandatory. If I have a non-expiring token, will it have a `ttl:""` present?

If not, probably it is better to just return false and no error here. (similar treatment as if it is a batch token)

Reply:
wouldnt we need to add an exception for if the ttlInt ==0 ? As in, this is the case where the call returned OK as being a root token. Wouldn't that cause the root token to always fail the check condition?

Or did you test that already? 😄

Reply:
I'm thinking about users with both:
```
apiVersion: external-secrets.io/v1beta1
kind: SecretStore
metadata:
  name: token
spec:
  provider:
    vault:
      server: "http://my.vault.server:8200"
      path: "secret"
      version: "v2"
      auth:
        tokenSecretRef:
          name: "vault-token"
          key: "token"
---
apiVersion: external-secrets.io/v1beta1
kind: SecretStore
metadata:
  name: jwt
spec:
  provider:
    vault:
      server: "http://my.vault.server:8200"
      path: "secret"
      auth:
         jwt: {...}
```
Wanting to use the feature because `jwt` SecretStore would benefit. However that would mean `token` SecretStore would break. Is my understanding correct?

Reply:
Let me just run some manual tests targeting your branch 😁 probably easiest for my brain to get around it

Reply:
I now did some tests to confirm it'll be an issue, but I've also found a fix for it.

Our edge case is whenever `creation_ttl == 0 && ttl == 0 `. This should be an allowed token :) 

I added a suggestion here for it, if you can add a test for it, this will LGTM 😄